File: retrofit/src/main/java/com/codekul/sonal/retrofit/MainActivity.java
package com.codekul.sonal.retrofit;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import com.codekul.sonal.retrofit.domain.DtoInfo;
import com.codekul.sonal.retrofit.domain.User;
import com.codekul.sonal.retrofit.rest.ApiService;
import com.codekul.sonal.retrofit.rest.RetrofitClient;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivity extends AppCompatActivity {

    public ApiService apiService;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        getRegister().setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                getPostData();
            }
        });
    }
    private void getPostData(){
        apiService= RetrofitClient.getClient().create(ApiService.class);
        Call<ResponseBody> call = apiService.userRegister(userObj());
        Log.i("@codekul" ,"regi url " +call.request().url().toString());
        call.enqueue(new Callback<ResponseBody>() {
            @Override
            public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
//                DtoInfo dtoInfo= response.body();
//                String message = dtoInfo.getMessage();
//                Log.i("@codekul"," resp : " +message);
                Log.i("@codekul " ,"response " +response);


            }

            @Override
            public void onFailure(Call<ResponseBody> call, Throwable t) {

                Log.i("@codekul","error msg " +call + " thro " + t.getMessage());

            }
        });

    }
    private User userObj(){
        User user = new User();
        user.setName(getName().getText().toString());
        user.setEmailId(getEmailId().getText().toString());
        user.setPassword(getPassword().getText().toString());
        user.setCity(getCity().getText().toString());
        return user;
    }



    private EditText getName(){
        return (EditText)findViewById(R.id.editName);
    }
    private EditText getPassword(){
        return (EditText)findViewById(R.id.editPassword);
    }
    private EditText getEmailId(){
        return (EditText)findViewById(R.id.editEmailId);
    }
    private EditText getCity(){
        return (EditText)findViewById(R.id.editCity);
    }
    private Button getRegister(){
        return (Button)findViewById(R.id.register);
    }



}

File: retrofitkotlin/src/main/java/com/codekul/sonal/retrofitkotlin/MainActivity.kt
package com.codekul.sonal.retrofitkotlin

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.EditText
import com.codekul.sonal.retrofitkotlin.R.id.*
import com.codekul.sonal.retrofitkotlin.domain.User
import com.codekul.sonal.retrofitkotlin.rest.ApiService
import kotlinx.android.synthetic.main.activity_main.*
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

abstract class MainActivity : AppCompatActivity() {

    lateinit var apiService: ApiService
    lateinit var editName: EditText
    lateinit var editEmail: EditText
    lateinit var editPassword: EditText
    lateinit var editCity: EditText
    lateinit var editOtpno :EditText

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

         editName = findViewById(R.id.editName)
         editEmail = findViewById(R.id.editEmailId)
        editPassword = findViewById(R.id.editPassword)
         editCity = findViewById(R.id.editCity)
        editOtpno = findViewById(R.id.editOtpno)

        register.setOnClickListener {
            getPostData()
        }
    }

    private fun getPostData() {
        apiService = ApiService.create()

        val call = apiService.userRegister(User(editName.text.toString(),editEmail.text.toString(),editPassword
                .text.toString(),editCity.text.toString(),editOtpno.text.toString()))

        Log.i("@codekul", """url = ${call.request().url()}""")
        call.enqueue(object : Callback<ResponseBody> {
            override fun onResponse(call: Call<ResponseBody>, response: Response<ResponseBody>) {
                //                DtoInfo dtoInfo= response.body();
                //                String message = dtoInfo.getMessage();
                //                Log.i("@codekul"," resp : " +message);
                Log.i("@codekul ", "response " + response)
            }

            override fun onFailure(call: Call<ResponseBody>, t: Throwable) {
                Log.i("@codekul", "error msg " + call + " thro " + t.message)
            }
        })
    }
}



